src/plugins/intel_gna/src/gna_plugin.cpp

```diff
@@ -1778,11 +1801,26 @@ std::vector<std::shared_ptr<const ov::Node>> GNAPlugin::GetOutputs() {
 }
 
 InferenceEngine::QueryNetworkResult GNAPlugin::QueryNetwork(const InferenceEngine::CNNNetwork& network,
-                                                            const std::map<std::string, std::string>& config) const {
+                                                            const std::map<std::string, std::string>& config_map) const {
     InferenceEngine::QueryNetworkResult res;
 
-    if (network.getFunction()) {
-        IE_THROW(NotImplemented) << " ngraph::Function is not supported natively";
+    GNAPluginNS::Config qn_config = config;
```

Comment:
Looks like a typo, config is renamed to config_map.
One one way to add one more Config constructor:
Config(std::map<std::string, std::string>& config_map)

Reply:
> Config

The idea was to get current config (which can be set before query network) and update with properties from query network itself